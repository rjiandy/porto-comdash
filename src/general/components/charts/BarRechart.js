// @flow

import React from 'react';
import {
  ResponsiveContainer,
  BarChart,
  XAxis,
  YAxis,
  Tooltip,
  Legend,
  Bar,
  // CartesianGrid,
} from 'recharts';
import {StyleSheet} from 'react-primitives';

import {View, Text} from '../coreUIComponents';
import {WHITE, GREY} from '../../constants/colors';
import {SMALL_FONT_SIZE} from '../../constants/text';
import {
  DEFAULT_BAR_SIZE,
  ANIMATION_DURATION,
} from '../../constants/chartSettings';

import type {ChartDatum} from '../../types/Chart-types';

type Props = {
  data: Array<ChartDatum>;
  xAxis: string;
  yAxis: string;
  showAxis?: boolean;
  showLabelValue?: boolean;
  showTooltip?: boolean;
  showLegend?: boolean;
  showChartGrid?: boolean;
  horizontal?: boolean;
  tickLabelPosition?: 'left' | 'right';
  width?: number;
  height?: number;
  style?: {
    bar?: {[key: string]: *};
    container?: {[key: string]: *};
    axis?: {[key: string]: *};
  };
  extraProps?: {
    container?: {[key: string]: *};
    xAxis?: {[key: string]: *};
    yAxis?: {[key: string]: *};
    barChart?: {[key: string]: *};
    bar?: {[key: string]: *};
    legend?: {[key: string]: *};
    tooltip?: {[key: string]: *};
  };
};

export default function BarRechart(props: Props) {
  let {
    data,
    xAxis,
    yAxis,
    showAxis,
    showLabelValue,
    showTooltip,
    showLegend,
    horizontal,
    tickLabelPosition,
    width,
    height,
    style,
    // showChartGrid,
  } = props;
  let extraProps = props.extraProps || {};
  let customStyle = style || {};
  return (
    <ResponsiveContainer
      width={width || '100%'}
      height={height || '100%'}
      {...extraProps.container}
    >
      <BarChart
        data={data}
        // NOTE: Layout is about the group names' position, NOT the bar's direction. Example:
        //                     |_ _ _
        //                   a |_ _ _|
        //                     |_ _
        //                   b |_ _|
        //                     |_ _ _ _
        //                   c |_ _ _ _|
        //                     | _ _ _ _ _
        //                       1 2 3 4 5
        //       [a, b, c] are the group names, while [1, 2, 3, 4, 5] are the values.
        //       The example is layout vertical because the group names are positioned vertically.
        layout={horizontal ? 'vertical' : 'horizontal'}
        {...extraProps.barChart}
      >
        {!horizontal ? (
          <XAxis
            type="category"
            dataKey={xAxis}
            tick={{fontSize: SMALL_FONT_SIZE, width: 150}}
            tickLine={false}
            axisLine={showAxis || false}
            interval={0}
            {...extraProps.xAxis}
          />
        ) : (
          <XAxis
            hide
            type="number"
            tick={false}
            tickLine={false}
            axisLine={false}
            {...extraProps.xAxis}
          />
        )}
        {horizontal ? (
          <YAxis
            type="category"
            width={120}
            dataKey={xAxis}
            orientation={tickLabelPosition || 'left'}
            tick={{fontSize: SMALL_FONT_SIZE, width: 150}}
            tickLine={false}
            axisLine={showAxis || false}
            interval={0}
            {...extraProps.yAxis}
          />
        ) : (
          <YAxis
            hide
            type="number"
            tick={false}
            tickLine={false}
            axisLine={false}
            {...extraProps.yAxis}
          />
        )}
        {data.length > 0 &&
          showTooltip && (
            <Tooltip
              content={(args) => <CustomTooltip {...args} />}
              {...extraProps.tooltip}
            />
          )}
        {showLegend && <Legend {...extraProps.legend} />}
        {/* {showChartGrid && <CartesianGrid strokeDasharray="3 3" />} */}
        <Bar
          animationDuration={ANIMATION_DURATION}
          dataKey={yAxis}
          fill={style && style.bar && style.bar.data.fill}
          barSize={
            (style && style.bar && style.bar.data.width) || DEFAULT_BAR_SIZE
          }
          maxBarSize={
            (style && style.bar && style.bar.data.width) || DEFAULT_BAR_SIZE
          }
          label={(args) =>
            showLabelValue && (
              <CustomizedBarTick
                {...extraProps.bar}
                {...args}
                horizontal={horizontal}
                style={(customStyle.bar && customStyle.bar.label) || {}}
                tickLabelPosition={tickLabelPosition || 'left'}
              />
            )}
          {...extraProps.bar}
        />
      </BarChart>
    </ResponsiveContainer>
  );
}

type TickLabelProps = {
  x: number;
  y: number;
  value: string;
  key: string;
  width: number;
  height: number;
  offset: number;
  horizontal: boolean;
  tickPlacement: 'inside' | 'outside';
  tickLabelPosition: 'left' | 'right';
  labelFormatter?: (label: string) => string;
  style?: StyleSheetTypes;
};

function CustomizedBarTick(props: TickLabelProps) {
  const {
    x,
    y,
    value,
    width,
    height,
    offset,
    horizontal,
    labelFormatter,
    style,
    tickPlacement,
    tickLabelPosition,
  } = props;
  let label = value;
  let defaultTickPlacement = tickPlacement || 'inside';
  let dy = -10;
  let dx = width / 2;
  if (defaultTickPlacement === 'inside') {
    if (!horizontal && Math.abs(height) < 17) {
      return null;
    }
    if (horizontal && Math.abs(width) < 20) {
      return null;
    }
    dy = (height + offset * 1.5) / 2;
    // dx = width / 2;
  } else if (defaultTickPlacement === 'outside') {
    if (horizontal) {
      dy = (height + offset * 1.5) / 2;
      dx = width + offset * 1.7 + 5;
      if (tickLabelPosition === 'right') {
        dx -= 30;
      }
    }
  }

  let customStyle = style || {};

  return (
    <g transform={`translate(${x},${y})`}>
      <text
        dx={dx}
        dy={dy}
        textAnchor="middle"
        fill={defaultTickPlacement === 'outside' ? 'black' : WHITE}
        style={{fontSize: SMALL_FONT_SIZE, zIndex: 10, ...customStyle}}
      >
        {label && labelFormatter ? labelFormatter(label) : label}
      </text>
    </g>
  );
}

type TooltipProps = {
  payload: Array<{[key: string]: any}>;
  label: string;
  dataKeyLabel?: {[key: string]: string};
  valueFormatter?: (value: number) => number;
};

function CustomTooltip(props: TooltipProps) {
  let {payload, label, dataKeyLabel, valueFormatter} = props;
  if (payload == null) {
    return null;
  }
  return (
    <View style={styles.tooltipContainer}>
      <Text style={styles.tooltipTitle}>{label}</Text>
      {payload.map((data, index) => {
        let color = data.fill;
        if (typeof data.fill === 'function') {
          color = data.fill(data.payload);
        }
        return (
          <Text key={index} style={{color}}>
            {dataKeyLabel != null ? dataKeyLabel[data.dataKey] : data.dataKey}:{' '}
            {valueFormatter ? valueFormatter(data.value) : data.value}
          </Text>
        );
      })}
    </View>
  );
}

const styles = StyleSheet.create({
  tooltipContainer: {
    backgroundColor: WHITE,
    borderColor: GREY,
    borderWidth: 1,
    padding: 10,
  },
  tooltipTitle: {
    fontWeight: 'bold',
    marginBottom: 5,
  },
});
