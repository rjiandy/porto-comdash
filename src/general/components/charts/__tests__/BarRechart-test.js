// @flow

import React from 'react';
import toJSON from 'enzyme-to-json';

import {shallow} from 'enzyme';

import BarRechart from '../BarRechart';

const data = [
  {name: 'a', value: 25},
  {name: 'b', value: 15},
  {name: 'c', value: 5},
  {name: 'd', value: 55},
];
describe('BarRechart', () => {
  it('Should render bar chart correctly', () => {
    let barChart = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" />
    );
    expect(toJSON(barChart)).toMatchSnapshot();
  });

  it('Should render horizontal bar chart correctly', () => {
    let barChartHorizontal = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" horizontal />
    );
    expect(toJSON(barChartHorizontal)).toMatchSnapshot();
  });

  it('Should render bar chart axis correctly', () => {
    let barChartWithAxis = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" showAxis />
    );
    expect(toJSON(barChartWithAxis)).toMatchSnapshot();
    let barChartHorizontalWithAxis = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" horizontal showAxis />
    );
    expect(toJSON(barChartHorizontalWithAxis)).toMatchSnapshot();
  });

  it('Should render bar chart tooltip correctly', () => {
    let barChartWithTooltip = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" showTooltip />
    );
    expect(toJSON(barChartWithTooltip)).toMatchSnapshot();
  });

  it('Should render bar chart legend correctly', () => {
    let barChartWithLegend = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" showLegend />
    );
    expect(toJSON(barChartWithLegend)).toMatchSnapshot();
  });

  it('Should render bar chart grid correctly', () => {
    let barChartWithLegend = shallow(
      <BarRechart data={data} yAxis="value" xAxis="name" showChartGrid />
    );
    expect(toJSON(barChartWithLegend)).toMatchSnapshot();
  });

  it('Should render bar chart axis ticks correctly', () => {
    let barChartWithTickLabelPosition = shallow(
      <BarRechart
        data={data}
        yAxis="value"
        xAxis="name"
        tickLabelPosition="right"
      />
    );
    expect(toJSON(barChartWithTickLabelPosition)).toMatchSnapshot();
    let barChartHorizontalWithTickLabelPosition = shallow(
      <BarRechart
        data={data}
        yAxis="value"
        xAxis="name"
        horizontal
        tickLabelPosition="right"
      />
    );
    expect(toJSON(barChartHorizontalWithTickLabelPosition)).toMatchSnapshot();
  });
});
