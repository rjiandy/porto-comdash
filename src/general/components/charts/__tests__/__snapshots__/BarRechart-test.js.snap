// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarRechart Should render bar chart axis correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={true}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      interval={0}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={true}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart axis correctly 2`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="vertical"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={true}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={true}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      interval={0}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={120}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart axis ticks correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      interval={0}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={true}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart axis ticks correctly 2`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="vertical"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={true}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      interval={0}
      mirror={false}
      orientation="right"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={120}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      interval={0}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={true}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart grid correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      interval={0}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={true}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart legend correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      interval={0}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={true}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Legend
      align="center"
      iconSize={14}
      layout="horizontal"
      verticalAlign="bottom"
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render bar chart tooltip correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      interval={0}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={true}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Tooltip
      active={false}
      animationDuration={400}
      animationEasing="ease"
      content={[Function]}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      isAnimationActive={true}
      itemSorter={[Function]}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;

exports[`BarRechart Should render horizontal bar chart correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "name": "a",
          "value": 25,
        },
        Object {
          "name": "b",
          "value": 15,
        },
        Object {
          "name": "c",
          "value": 5,
        },
        Object {
          "name": "d",
          "value": 55,
        },
      ]
    }
    layout="vertical"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={true}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={false}
      tickCount={5}
      tickLine={false}
      type="number"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      axisLine={false}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      interval={0}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tick={
        Object {
          "fontSize": 11,
          "width": 150,
        }
      }
      tickCount={5}
      tickLine={false}
      type="category"
      width={120}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={500}
      animationEasing="ease"
      barSize={30}
      data={Array []}
      dataKey="value"
      hide={false}
      isAnimationActive={true}
      label={[Function]}
      layout="vertical"
      legendType="rect"
      maxBarSize={30}
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</ResponsiveContainer>
`;
